BlobStorage/Advanced.cs

```diff
@@ -736,24 +650,25 @@ private static async Task ManageContainerLeasesAsync(CloudBlobClient blobClient)
                     Case 5: Attempt to delete leased container without lease ID.
                 */
 
-                container5 = blobClient.GetContainerReference(LeasingPrefix + Guid.NewGuid());
+                container5 = blobServiceClient.GetBlobContainerClient(LeasingPrefix + Guid.NewGuid());
                 await container5.CreateIfNotExistsAsync();
-                
+
                 // Acquire the lease.
-                await container5.AcquireLeaseAsync(leaseDuration, null);
+                var leaseContainer5 = container1.GetBlobLeaseClient();
+                var containerLease5 = await leaseContainer1.AcquireAsync(leaseDuration);
+                leaseId = containerLease5.Value.LeaseId;
 
                 // Get container properties to see that the container has been leased.
-                await container5.FetchAttributesAsync();
                 PrintContainerLeaseProperties(container5);
 
                 // Attempt to delete the leased container without the lease ID.
                 // This operation will result in an error.
                 // Note that in a real-world scenario, it would most likely be another client attempting to delete the container.
                 await container5.DeleteAsync();
             }
-            catch (StorageException e)
+            catch (RequestFailedException e)
             {
-                if (e.RequestInformation.HttpStatusCode == 412)
+                if (e.Status == 412)
```

Comment:
We should also check what the code is when the error handling is this specific:
```suggestion
                if (e.Status == 412 && e.ErrorCode == BlobErrorCode.LeaseIdMissing)
```
(I think it should be `LeaseIdMissing`, but it'd be good to run this and check.)